server/src/main/java/org/elasticsearch/action/ActionType.java

```diff
@@ -27,8 +49,17 @@ public static ActionType<ActionResponse.Empty> emptyResponse(String name) {
     }
 
     /**
-     * @param name The name of the action, must be unique across actions.
-     * @param responseReader A reader for the response type
+     * Construct an {@link ActionType} which callers can execute both on the local node (using {@link NodeClient}) and on a remote cluster
+     * (using a client obtained from {@link Client#getRemoteClusterClient}). If the action is only to be executed on the local cluster then
+     * declare it using {@link #localOnly} instead.
+     * <p>
+     * There is no facility for directly executing an action on a different node in the local cluster. To achieve this, implement an action
+     * which runs on the local node and knows how to use the {@link TransportService} to forward the request to a different node. There are
+     * several utilities that help implement such an action, including {@link TransportNodesAction} or {@link TransportMasterNodeAction}.
+     *
+     * @param name           The name of the action, which must be unique across actions. When executed on a remote cluster, this is the
+     *                       ID of the transport action which is sent to the handling node in the remote cluster.
+     * @param responseReader A reader for the response type, which is only used by actions executed on a remote cluster.
```

Comment:
> A reader for the response type, which is only used by actions executed on a remote cluster.

Is that a reader for the response from the action executed on the remote cluster? I'm trying to read the as-written words incorrectly and I feel like I can convince myself that those words say that it's the action executed on the remote cluster that uses the reader.

Reply:
Fair, I see the ambiguity. Let me rearrange the words, sec.

Reply:
1b8526bb9659 any better?

Reply:
SO MUCH BETTER. Thank you! I love it!